src/main/java/com/vackosar/gitflowincrementalbuild/control/DifferentFiles.java

```diff
@@ -220,4 +257,145 @@ private boolean pathNotExcluded(Path path) {
+
+            // add additional environment entries, if present (test only)
+            if (!additionalNativeGitEnvironment.isEmpty()) {
+                procBuilder.environment().putAll(additionalNativeGitEnvironment);
+            }
+            procBuilder.directory(projectDir.toFile());
+
+            ExecutionResult result = fs.execute(procBuilder, new ByteArrayInputStream(buildGitCommandInput(uri).getBytes()));
+            if (result.getRc() != 0) {
+                logger.info(bufferToString(result.getStdout()));
+                logger.error(bufferToString(result.getStderr()));
+                throw new IllegalStateException("Native Git invocation failed with return code " + result.getRc()
+                        + ". See previous log output for more details.");
+            }
+
+            return extractCredentials(bufferToString(result.getStdout()));
+        }
+
+        // build input for "git credential fill" as per https://git-scm.com/docs/git-credential#_typical_use_of_git_credential
+        private String buildGitCommandInput(URIish uri) {
+            StringBuilder builder = new StringBuilder();
+            builder.append("protocol=").append(uri.getScheme()).append("\n");
+            builder.append("host=").append(uri.getHost());
+            if (uri.getPort() != -1) {
+                builder.append(":").append(uri.getPort());
+            }
+            builder.append("\n");
+            Optional.ofNullable(uri.getPath())
+                    .map(path -> path.startsWith("/") ? path.substring(1) : path)
+                    .ifPresent(path -> builder.append("path=").append(path).append("\n"));
+            Optional.ofNullable(uri.getUser())
+                    .ifPresent(user -> builder.append("username=").append(user).append("\n"));
+            return builder.toString();
+        }
+
+        private String bufferToString(TemporaryBuffer buffer) throws IOException {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            buffer.writeTo(baos, null);
+            return baos.toString();
+        }
+
+        private CredentialsPair extractCredentials(String nativeGitOutput) {
+            Matcher matcher = Pattern.compile("(?<=username=).+|(?<=password=).+").matcher(nativeGitOutput);
+            if (!matcher.find()) {
+                throw new IllegalStateException("Could not find username in native Git output");
+            }
+            String username = matcher.group();
+            if (!matcher.find()) {
+                throw new IllegalStateException("Could not find password in native Git output");
+            }
+            char[] password = matcher.group().toCharArray();
+
+            CredentialsPair credPair = new CredentialsPair();
+            credPair.username = username;
+            credPair.password = password;
+            return credPair;
+        }
+
+        private class CredentialsPair {
+            String username;
+            char[] password;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/gitflow-incremental-builder/gitflow-incremental-builder/pullRequest?prid=4845076)